### PR TITLE
kiwix-serve integration in kiwix-lib

### PR DESCRIPTION
**Pull request by jetownfeve21**

### CODE REVIEW

**Comment by jetownfeve21** [comment]
@mgautierfr 
I moved the subprocess.h into the "include" directory to compile but I don't know if it's the right way to do it

**Comment by kelson42** [comment]
It is really important attachProcessOpt works fine

**Comment by jetownfeve21** [comment]
Right now I delete the process with the kill() method of the "SubProcess" class instead

**Comment by kelson42** [comment]
AFAIK all the problems to do so have already been fixed in https://github.com/kiwix/kiwix-xulrunner/blob/master/src/components/serverManager/serverManager.cpp

**Comment by jetownfeve21** [comment]
Just add ' ' around the pid string solved the problem

File: include/meson.build [hunk 1]
```diff
@@ -9,7 +9,9 @@ headers = [
   'downloader.h',
   'reader.h',
   'entry.h',
-  'searcher.h'
+  'searcher.h',
+  'kiwixserve.h',
+  'subprocess.h'
```
Comment (mgautierfr) [inline include/meson.build]: This is the list of header files to install.
Header we don't want to install (not part of the public API) don't need (and must not) to be installed.
So we don't need `subprocess.h` here.

File: include/kiwixserve.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#ifndef KIWIXLIB_KIWIX_SERVE_H_
+#define KIWIXLIB_KIWIX_SERVE_H_
+
+#ifdef _WIN32
+// winsock2.h need to be included before windows.h (included by curl.h)
+# include <winsock2.h>
+#endif
+
+#include "subprocess.h"
```
Comment (mgautierfr) [inline include/kiwixserve.h]: We must not include subprocess here. This is a private header.
You can simply declare the `Subprocess` class and include the `subprocess.h` header in the cpp file.

**Pull request merged**